src/Model/Filter/Like.php

```diff
@@ -143,26 +155,34 @@ protected function _wildcards($value)
      */
     protected function _formatWildcards($value)
     {
-        $from = $to = $substFrom = $substTo = [];
-        if ($this->config('wildcardAny') !== '%') {
-            $from[] = '%';
-            $to[] = '\%';
-            $substFrom[] = $this->config('wildcardAny');
-            $substTo[] = '%';
-        }
-        if ($this->config('wildcardOne') !== '_') {
-            $from[] = '_';
-            $to[] = '\_';
-            $substFrom[] = $this->config('wildcardOne');
-            $substTo[] = '_';
+        $value = $this->_escapeDriver->formatWildcards($value);
+
+        return $value;
+    }
+
+    /**
+     * set configuration for escape driver name
+     *
+     * @return void
+     */
+    protected function _setEscapeDriver()
+    {
+        if ($this->config('escapeDriver') === null) {
+            $driver = get_class($this->query()->connection()->driver());
+            $driverName = 'Sqlserver';
+            if (substr_compare($driver, $driverName, -strlen($driverName)) === 0) {
+                $this->config('escapeDriver', 'Search.Sqlserver');
+            } else {
+                $this->config('escapeDriver', 'Search.Default');
+            }
         }
-        if ($from) {
-            // Escape first
-            $value = str_replace($from, $to, $value);
-            // Replace wildcards
-            $value = str_replace($substFrom, $substTo, $value);
+
+        $class = $this->config('escapeDriver');
+        $className = App::className($class, 'Model/Filter/EscapeDriver', 'Escaper');
+        if (!class_exists($className)) {
```

Comment:
`App::className` already does a `class_exists()` check and returns `false` on failure, so this should be just `if (!$className) {`